timeseries/src/autogluon/timeseries/models/local/statsforecast.py

```diff
@@ -391,3 +415,343 @@ def _get_model_type(self):
+            **{str(q): point_forecast + q_sign[i] * ehat[i] for i, q in enumerate(levels)},
+        }
+        return pd.DataFrame(predictions)
+
+
+class AutoCESModel(AbstractConformalizedStatsForecastModel):
+    """Forecasting with an Complex Exponential Smoothing model where the model selection is performed using the
+    Akaike Information Criterion.
+
+    Based on `statsforecast.models.AutoCES <https://nixtla.mintlify.app/statsforecast/docs/models/autoces.html>`_.
+
+
+    References
+    ----------
+    .. [Svetunkov2022] Svetunkov, Ivan, Nikolaos Kourentzes, and John Keith Ord. "Complex exponential
+        smoothing." Naval Research Logistics (NRL) 69.8 (2022): 1108-1123.
+
+
+    Other Parameters
+    ----------------
+    model : str, default = "Z"
+        Defines type of CES model, "N" for simple CES, "S" for simple seasonality, "P" for partial seasonality
+        (without complex part), $F$ for full seasonality. When "Z" is selected, the best model is selected using
+        Akaike Information Criterion (AIC).
+    seasonal_period : int or None, default = None
+        Number of time steps in a complete seasonal cycle for seasonal models. For example, 7 for daily data with a
+        weekly cycle or 12 for monthly data with an annual cycle.
+        When set to None, seasonal_period will be inferred from the frequency of the training data. Can also be
+        specified manually by providing an integer > 1.
+        If seasonal_period (inferred or provided) is equal to 1, seasonality will be disabled.
+    n_jobs : int or float, default = 0.5
+        Number of CPU cores used to fit the models in parallel.
+        When set to a float between 0.0 and 1.0, that fraction of available CPU cores is used.
+        When set to a positive integer, that many cores are used.
+        When set to -1, all CPU cores are used.
+    max_ts_length : int, default = 2500
+        If not None, only the last ``max_ts_length`` time steps of each time series will be used to train the model.
+        This significantly speeds up fitting and usually leads to no change in accuracy.
+    """
+
+    allowed_local_model_args = [
+        "model",
+        "seasonal_period",
+    ]
+
+    def _get_model_type(self):
+        from statsforecast.models import AutoCES
+
+        return AutoCES
+
+    def _update_local_model_args(self, local_model_args: dict) -> dict:
+        local_model_args = super()._update_local_model_args(local_model_args)
+        local_model_args.setdefault("model", "Z")
+        return local_model_args
+
+    def _get_point_forecast(self, time_series: pd.Series, local_model_args: Dict):
+        # Disable seasonality if time series too short for chosen season_length or season_length == 1,
+        # otherwise model will crash
+        if len(time_series) < 5:
+            # AutoCES does not handle "tiny" datasets, fall back to naive
+            return np.full(self.prediction_length, time_series.values[-1])
```

Comment:
Related to above comments: If we change `AutoCES` to inherit from `AbstractProbabilisticStatsForecastModel`, we can use the `seasonal_naive_forecast` method from `autogluon.timeseries.models.local.abstract_local_model` here.